lib/fizzy/execute.cpp

```diff
@@ -840,7 +840,7 @@ ExecutionResult execute(Instance& instance, FuncIdx func_idx, const Value* args,
                     throw std::bad_alloc();
                 memory->resize(new_pages * PageSize);
             }
-            catch (std::bad_alloc const&)
+            catch (const std::exception&)
```

Comment:
Add a comment that `std::bad_alloc` and `std::legth_error` are expected.

Reply:
The ... catches everything so it guarantees noexcept outside. The std::exception does not, so there will be another compiler generated check ending in std::terminate or something.

So my final call is to use ... and also add an assert that the exception is one of the two expected types.

Although, checking the exception type may be tricky. Probably a helper is needed as presented in https://en.cppreference.com/w/cpp/error/rethrow_exception

Reply:
Well then in the helper `catch std::length_error` will still be not covered, it's simpler then to have here `catch (bad_alloc)` + `catch (length_error)` + `catch (...) { assert(false); } `.

Reply:
> Well then in the helper `catch std::length_error` will still be not covered, it's simpler then to have here `catch (bad_alloc)` + `catch (length_error)` + `catch (...) { assert(false); } `.

But yeah `execute` will be smaller if extra catches go into helper, and helper will should be omitted in release build...